### PR TITLE
Refactor for queries

### PR DESCRIPTION
**Pull request by kmagida110**

### CODE REVIEW

File: dashboard/server.R [hunk 1]
```diff
@@ -289,29 +218,26 @@ shinyServer(function(input, output) {
...
-      theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
-      labs(x = 'Topics', y = 'Conversion Rate (%)', title = 'Success Rate for Top Topics')
+    plt = topic.conversions %>% 
+            dplyr::inner_join(., topic.sessions,by = "topic_title") %>%
+            dplyr::mutate(conversion_rate = round(conversions / sessions,3)) %>%   
+            dplyr::arrange(., desc(sessions)) %>%
+            dplyr::filter(sessions > quantile(sessions, 1 - (as.numeric(input$pct.cutoffs) / 100))) %>% # Top X% based on sessions
+            ggplot(., aes_string(x = RATE.GROUP.COL, y = 'conversion_rate')) +
+            geom_bar(stat = 'identity') + 
+            theme_bw() +
```
Comment (comckay) [inline dashboard/server.R]: make bars black so plots match

File: dashboard/ui.R [hunk 2]
```diff
@@ -32,22 +32,23 @@ shinyUI(navbarPage(
...
-                          "Event Type" = "event.type"),
-              selected = "all",
-              inline = T
+                          "Page Type" = PAGE.TYPE,
+                          "Service Type" = SERVICE.TYPE,
+                          "Event Type" = EVENT.TYPE),
+              selected = "all"
             ),
             uiOutput("type.selection.options"),
+            br(),
```
Comment (comckay) [inline dashboard/ui.R]: layout will be handled by minghua's changes so please remove these br() calls

**Comment by comckay** [comment]
screen shot 2017-02-23 at 1 36 58 pm
screen shot 2017-02-23 at 1 37 03 pm

File: dashboard/functions/helper.R [hunk 3]
```diff
@@ -105,10 +153,17 @@ groupAndOrder <- function(df, group.col, data.col, percent = TRUE,top.pct = 1, f
   if (top.pct > 1) {
     stop("top.pct cannot be > 1")
   }
+
+  if (top.pct < 1 & get.top.k == FALSE){
+    stop("filtering and taking from the bottom")
+  }
+ 
   if (class(df[[group.col]]) %in% c("factor", "character") == F) {
+    print(group.col)
```
Comment (kmagida110) [inline dashboard/functions/helper.R]: Set

File: dashboard/functions/helper.R [hunk 4]
```diff
@@ -87,7 +87,55 @@ flagIncompleteTimeperiod <- function(reference.vector, time.unit) {
...
+
+ groups = df %>% dplyr::group_by_(group.col) %>%
+            dplyr::rename_(n = paste(data.col), group = paste(group.col)) %>%
+            dplyr::summarise(n = sum(n)) %>%
+            dplyr::arrange(dplyr::desc(n))
+ 
+ if(k > nrow(groups)){
+  
+   k = nrow(groups)
+
```
Comment (kmagida110) [inline dashboard/functions/helper.R]: Done

File: dashboard/functions/helper.R [hunk 5]
```diff
@@ -87,7 +87,55 @@ flagIncompleteTimeperiod <- function(reference.vector, time.unit) {
...
+         slice.from = 1
+         slice.to = k
+       } else{
+         slice.from = nrow(groups) - k + 1
+         slice.to = nrow(groups)
+       }
+ 
+  top.groups = groups %>% 
+                    dplyr::slice(slice.from:slice.to) %>%
+                    select(group) %>%
```
Comment (kmagida110) [inline dashboard/functions/helper.R]: Done

File: dashboard/get_data/query.R [hunk 6]
```diff
@@ -3,7 +3,8 @@ library(DBI)
 library(RPostgreSQL)
 
 # Source creds
-source("dashboard/get_data/db_connect.R")
+source("db_connect.R")
+source("../functions/constants.R")
```
Comment (kmagida110) [inline dashboard/get_data/query.R]: leaving as is for now, assumes run in dashboard

File: dashboard/get_data/query.R [hunk 7]
```diff
@@ -12,119 +13,41 @@ db.connection <- dbConnect(db.driver, dbname = db.name,
...
+paste("SELECT parent.content_type, 
+      parent.title, 
+      d.descendant_id
+      FROM drupal_node_descendants_new as d
+      INNER JOIN drupal_nodes AS parent ON d.node_id = parent.node_id") %>%
+      RPostgreSQL::dbGetQuery(statement = ., conn = db.connection) %>%
+      {
+        saveRDS(., "../data/drupal_node_descendants.RDS")
+        drupal.node.descendants <<- . 
+      }
```
Comment (kmagida110) [inline dashboard/get_data/query.R]: Done

File: dashboard/server.R [hunk 8]
```diff
@@ -1,136 +1,117 @@
...
   shinyURL.server()
   #### VISITOR SUCCESS ####
   # DATA MUNGING
   # subset by timeseries start and end
   visitor.success.subset.timeseries <- reactive({  # creates data frame subset by time selection
     ga.conversions %>%
-      dplyr::filter(lubridate::date(hit_timestamp_eastern) >= input$visitor.success.daterange[1],
-                    lubridate::date(hit_timestamp_eastern) <= input$visitor.success.daterange[2])
+      dplyr::filter(lubridate::date(date) >= input$visitor.success.daterange[1],
+                    lubridate::date(date) <= input$visitor.success.daterange[2])
```
Comment (kmagida110) [inline dashboard/server.R]: Done

File: dashboard/server.R [hunk 9]
```diff
@@ -1,136 +1,117 @@
...
     }
+    
+    # Group dataframe from selected, always including date
+    if (is.null(input$visitor.success.group.by)) {
+        
+        group.vars = c('date')
+      
+      } else{ 
+        
+        renamed.cols = unlist(lapply(input$visitor.success.group.by, FUN = function(x) {gsub(' ','_',x)}))
```
Comment (kmagida110) [inline dashboard/server.R]: Example?

Comment (comckay) [inline dashboard/server.R]: `purrr::map(input$visitor.success.group.by, function(x) {gsub(pattern = " ", replacement = "-", x)}`

Comment (kmagida110) [inline dashboard/server.R]: Did that, got a weird error

**Pull request merged**